src/main/cpp/VideoState.cpp

```diff
@@ -149,6 +149,15 @@ int VideoState::stream_component_open(int stream_index) {
 		this->video_stream = stream_index;
 		this->video_st = ic->streams[stream_index];
 
+		// Calculate the Frame rate (FPS) of the video stream
+		if (this->video_st) {
+			AVRational rational = this->video_st->avg_frame_rate;
+			if(rational.den == rational.num == 0)
+				rational = this->video_st->r_frame_rate;
+
+			this->fps = rational.num / rational.den;
```

Comment:
Why not use?
AVRational fr = av_guess_frame_rate(ic, video_st, NULL);
fps = av_q2d(fr);